grpc/src/main/java/io/stargate/grpc/codec/cql/IntCodec.java

```diff
@@ -30,18 +29,18 @@ public ByteBuffer encode(@NonNull QueryOuterClass.Value value, @NonNull Column.C
     if (value.getInnerCase() != InnerCase.INT) {
       throw new IllegalArgumentException("Expected integer type");
     }
-    try {
-      return TypeCodecs.INT.encodePrimitive(
-          Math.toIntExact(value.getInt()), ProtocolVersion.DEFAULT);
-    } catch (ArithmeticException e) {
-      throw new IllegalArgumentException("Integer overflow", e);
+    int intValue = (int) value.getInt();
```

Comment:
Why not have a dedicated `Value` field for `int` to avoid overflow?

Reply:
gRPC uses variable encoded integers and there's no type for `short` or `byte` so I think it makes sense to a have a single integer value that can represent `tinyint`, `smallint`, `int` and `bigint`. It also helps minimize the number of fields per the last comment.

Note: We should likely be using `sint64` for this type as per: https://developers.google.com/protocol-buffers/docs/encoding#signed_integers

Reply:
Ha, I was already using `sint64` :)